packages/block-library/src/site-logo/index.php

```diff
@@ -58,6 +76,59 @@ function render_block_core_site_logo( $attributes ) {
 	return $html;
 }
 
+/**
+ * Generates class names and styles to apply the border support styles for
+ * the site logo block.
+ *
+ * @param array $attributes The block attributes.
+ * @return array The border-related classnames and styles for the block.
+ */
+function block_core_site_logo_get_border_attributes( $attributes ) {
+
+	$border_styles = array();
+	$sides         = array( 'top', 'right', 'bottom', 'left' );
+
+	// Border radius.
+	if ( isset( $attributes['style']['border']['radius'] ) ) {
+		$border_styles['radius'] = $attributes['style']['border']['radius'];
+	}
+
+	// Border style.
+	if ( isset( $attributes['style']['border']['style'] ) ) {
+		$border_styles['style'] = $attributes['style']['border']['style'];
+	}
+
+	// Border width.
+	if ( isset( $attributes['style']['border']['width'] ) ) {
+		$border_styles['width'] = $attributes['style']['border']['width'];
+	}
+
+	// Border color.
+	$preset_color           = array_key_exists( 'borderColor', $attributes ) ? "var:preset|color|{$attributes['borderColor']}" : null;
+	$custom_color           = _wp_array_get( $attributes, array( 'style', 'border', 'color' ), null );
```

Comment:
```suggestion
	$custom_color           = $attributes['style']['border']['color'] ?? null;
```

From a performance standpoint, `_wp_array_get()` function may be best avoided. See #51116